### ISSUE TITLE
Superstore S02E02 keeps getting identified as S02E01

### ISSUE THREAD

**Issue by G33K** [pymedusa/Medusa#10387]
**Describe the bug**
I have only seen this happen with Superstore S02E02 so far.
Superstore S02E02 gets parsed and identified as S02E01 and then the actual S02E01 is overwritten with what is actually S02E02. I have tried different file name formats with and without the quality and even just naming it as Superstore S02E02.mkv gets the same results. Tried toggling Scene Numbering setting with same result and the scene values are set to correct 2x1, 2x2

If I manually rename and place the correct S02E02 in the folder and do a full update on the series, the file stays but the S02E02 nfo and thumbs generated are still for S02E01 and the S02E02 row in the show table does not list the filename.

I have tried Automated Post processing, and manual post processing with same result.

I have tried restarting Medusa, restarting the server, removing the cache folder and the cache.db file (shutdown, remove, restart) but still same outcome. From the logs it looks like it might be a cache issue but which other cache is there?

2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Parse result (release_group): None
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Parse result (air_date): None
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Parse result (ab_episode_numbers): []
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Parse result (episode_numbers): [1]
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Parse result (season_number): 2
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Parse result (series_name): Superstore
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Parse result not sufficient (all following have to be set). Won't save release name
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Parsed Superstore.S02E02.mkv into title: Superstore, season: 2, episode: [1], container: mkv, mimetype: video/x-matroska, type: episode, parsing_time: 0.010959625244140625, absolute_episode: [], quality: Unknown, total_time: 0.017665863037109375
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Matched release Superstore.S02E02.mkv to a series in your database: Superstore
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Using guessit cache item for Superstore.S02E02.mkv
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Analyzing name: Superstore.S02E02.mkv
2022-03-03 07:16:33 INFO	Thread-2 :: [4908f30] Processing /.../Superstore.S02E02.mkv
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Post-processing video files: ['Superstore.S02E02.mkv']
2022-03-03 07:16:33 DEBUG	Thread-2 :: [4908f30] Post-processing files: ['Superstore.S02E02.mkv']

Slightly different log with different filename:

2022-03-03 07:24:35 DEBUG	Thread-30 :: [4908f30] Retrieving episode object for Superstore S02E01
2022-03-03 07:24:35 DEBUG	Thread-30 :: [4908f30] Show information parsed from file name
2022-03-03 07:24:35 DEBUG	Thread-30 :: [4908f30] Using cache item for Superstore S02E02 1080p.mkv
2022-03-03 07:24:35 DEBUG	Thread-30 :: [4908f30] Analyzing name: Superstore S02E02 1080p.mkv
2022-03-03 07:24:35 INFO	Thread-30 :: [4908f30] Processing /.../Superstore S02E02 1080p.mkv
2022-03-03 07:24:35 DEBUG	Thread-30 :: [4908f30] Post-processing video files: ['Superstore S02E02 1080p.mkv']
2022-03-03 07:24:35 DEBUG	Thread-30 :: [4908f30] Post-processing files: ['Superstore S02E02 1080p.mkv']

**Comment by p0psicles** [comment]
What does the guessit test tool provide you with? It's in config - postprocessing

**Comment by G33K** [comment]
guessit tool always returns the correct episode number regardless of how the filename is formatted.

**Comment by p0psicles** [comment]
Do you have the scene option enabled for the show.
Or did you overwrite the scene episode numbering ?

**Comment by G33K** [comment]
Scene option is off but I've tried toggling it on and still got the same result. The scene episode numbering is also correctly set to 2x1 and 2x2 for the respective episodes.

**Comment by p0psicles** [comment]
When the scene option is disabled. A restart of Medusa would reset any name parse caching.

So so that. Then enable debug logging and process one episode. Share that log.

**Comment by G33K** [comment]
So, I did that. Made sure scene option was disabled and restarted and tried processing S02E02 with the same result, gets detected as S02E01.

However, this time I noticed this line in the log:

Found local numbering S02E02 from scene S02E02 for show Superstore

So this got me thinking it has to do something with scene numbering even though its off.

After a lot of trial and error this is what fixed it:

- Turn on scene numbering
- Changed scene numbering values for S02E01 and S02E02, to some random values
- Changed scene numbering values again back to the original and correct values
- Turn off scene numbering
- Restart (just in case)

Finally got  it correctly processed.

So, even though scene numbering was off it was still interfering in some way even though I don't remember ever messing around with the scene numbers to begin with.
